From 36e33add76dd6f7856ecf1af317bd95d115926dc Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Mon, 7 Jul 2014 01:05:03 +0300
Subject: [PATCH] Changed to cuboxi device config

1. reverted back to LK 3.0.35
2. audio sampling rate to 48k fixed
3. changed build rev to indicate 4.4.2-1.0.0

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 cuboxi/BoardConfig.mk    |   20 ++++++++++++++++----
 cuboxi/audio_policy.conf |    6 +++---
 cuboxi/build_id.mk       |    4 ++--
 cuboxi/init.rc           |    4 ++++
 4 files changed, 25 insertions(+), 9 deletions(-)

diff --git a/cuboxi/BoardConfig.mk b/cuboxi/BoardConfig.mk
index e198213..5e0ca37 100755
--- a/cuboxi/BoardConfig.mk
+++ b/cuboxi/BoardConfig.mk
@@ -70,8 +70,7 @@ $(error "TARGET_USERIMAGES_USE_UBIFS and TARGET_USERIMAGES_USE_EXT4 config open
 endif
 endif
 
-BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init gpumem=96M video=mxcfb0:dev=hdmi,1280x720M@60,if=RGB24 video=mxcfb1:off video=mxcfb2:off fbmem=10M vmalloc=400M androidboot.console=ttymxc0 androidboot.hardware=freescale
-#BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init video=mxcfb0:dev=ldb,bpp=32 video=mxcfb1:off video=mxcfb2:off fbmem=10M fb0base=0x27b00000 vmalloc=400M androidboot.console=ttymxc0 androidboot.hardware=freescale
+BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init video=mxcfb0:dev=hdmi,1280x720M@60,if=RGB24 video=mxcfb1:off video=mxcfb2:off fbmem=10M vmalloc=400M androidboot.console=ttymxc0 androidboot.hardware=freescale
 
 ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
 #UBI boot command line.
@@ -93,7 +92,20 @@ BOARD_HAVE_USB_CAMERA := true
 # define frame buffer count
 NUM_FRAMEBUFFER_SURFACE_BUFFERS := 3
 
+#define consumer IR HAL support
+#IMX6_CONSUMER_IR_HAL := true
 
 TARGET_BOOTLOADER_CONFIG := mx6_cubox-i_config
-TARGET_KERNEL_DEFCONF := imx_v7_cubox-i_hummingboard_android_defconfig
-TARGET_KERNEL_DTB := imx6q-cubox-i.dtb imx6dl-cubox-i.dtb imx6dl-hummingboard.dtb
+TARGET_KERNEL_DEFCONF := imx6_cubox-i_hummingboard_android_defconfig
+# For Linux kernel 3.10.x uncomment the following
+#TARGET_KERNEL_DEFCONF := imx_v7_cubox-i_hummingboard_android_defconfig
+#TARGET_KERNEL_DTB := imx6q-cubox-i.dtb imx6dl-cubox-i.dtb imx6dl-hummingboard.dtb
+BOARD_SEPOLICY_DIRS := \
+       device/fsl/sabresd_6dq/sepolicy
+
+BOARD_SEPOLICY_UNION := \
+       app.te \
+       file_contexts \
+       fs_use \
+       untrusted_app.te \
+       genfs_contexts
diff --git a/cuboxi/audio_policy.conf b/cuboxi/audio_policy.conf
index c709621..b61a0fe 100644
--- a/cuboxi/audio_policy.conf
+++ b/cuboxi/audio_policy.conf
@@ -5,7 +5,7 @@
 global_configuration {
   attached_output_devices AUDIO_DEVICE_OUT_SPEAKER
   default_output_device AUDIO_DEVICE_OUT_SPEAKER
-  attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC
+  attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC|AUDIO_DEVICE_IN_REMOTE_SUBMIX
 }
 
 # audio hardware module section: contains descriptors for all audio hw modules present on the
@@ -59,7 +59,7 @@ audio_hw_modules {
   r_submix {
     outputs {
       submix {
-        sampling_rates 44100|48000
+        sampling_rates 48000
         channel_masks AUDIO_CHANNEL_OUT_STEREO
         formats AUDIO_FORMAT_PCM_16_BIT
         devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
@@ -67,7 +67,7 @@ audio_hw_modules {
     }
     inputs {
       submix {
-        sampling_rates 44100|48000
+        sampling_rates 48000
         channel_masks AUDIO_CHANNEL_IN_STEREO
         formats AUDIO_FORMAT_PCM_16_BIT
         devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
diff --git a/cuboxi/build_id.mk b/cuboxi/build_id.mk
index 013f595..4a1b522 100644
--- a/cuboxi/build_id.mk
+++ b/cuboxi/build_id.mk
@@ -18,5 +18,5 @@
 # (like "CRB01").  It must be a single word, and is
 # capitalized by convention.
 
-export BUILD_ID=1.0.0-alpha-rc2
-export BUILD_NUMBER=20140110
+export BUILD_ID=1.0.0-rc3
+export BUILD_NUMBER=20140630
diff --git a/cuboxi/init.rc b/cuboxi/init.rc
index 137435a..8711483 100755
--- a/cuboxi/init.rc
+++ b/cuboxi/init.rc
@@ -93,6 +93,10 @@ on boot
 
     # Set the density to 160dpi, default 128dpi is not good
     setprop ro.sf.lcd_density 160
+
+    # Set extsd access permission
+    setprop persist.sampling_profiler 1
+
 # change for FSL specific service
     chown root system /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
     chmod 0664 /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
-- 
1.7.9.5

